gatsby/content/blog/2021/12/2021-12-03-sydent-typing-1.mdx

```diff
@@ -0,0 +1,187 @@
+---
+  date: '2021-12-03'
+  title: 'Type coverage for Sydent: motivation'
+  categories:
+    - Thoughts
+    - Tech
+  author: David Robertson
+---
+
+[Sydent](https://github.com/matrix-org/sydent) is the reference Matrix Identity server. It provides a [lookup service](https://matrix.org/docs/spec/identity_service/r0.3.0), so that you can find a Matrix user via their email address or phone number (if they've chosen to share it).
+
+We recently worked on [improving Sydent's type coverage](https://github.com/matrix-org/sydent/issues/414). Type coverage (or "typing") refers to annotations in Python source code that specify the type of an expression. These annotations are entirely optional, but if present, they allow tools to reason about the types that flow through a program. One popular tool is [`mypy`](https://mypy.readthedocs.io/en/stable/), which can often use the type information to spot bugs before you ship them! We worked to make Sydent pass `mypy --strict`. It now does so, with an overall precision of 94%! Here's what that process looks like:
```

Comment:
You say "type" a lot in this paragraph. Can it be cut down?

> Type coverage (or "typing") refers to annotations in Python source code that specify the type of an expression.

Close to being a self-referential definition ("typing refers to specifying types")...

As a rough sketch, maybe something along the lines of:

> We recently worked on improving Sydent's type coverage: the proportion of the codebase with explicit annotations denoting the kind of data that each variable, parameter, and return value is expected to hold. These annotations are optional, but when they're present tools like mypy can use them to analyze your code and prevent entire classes of errors _before_ you ship them! In this instance we attempted to make Sydent pass mypy in "strict" mode, which it now does. Here's what the process looked like:

Reply:
Thanks, I've nicked most of this and applied a little tweaking in dac5e78b.